src/storage/txn/process.rs

```diff
@@ -205,14 +213,13 @@ impl Task {
             "process cmd with snapshot, cid={}, cb_ctx={:?}",
             self.cid, cb_ctx
         );
-        let mut cmd = self.cmd.take().unwrap();
+        let tag = self.tag;
         if let Some(term) = cb_ctx.term {
-            cmd.mut_context().set_term(term);
+            self.cmd.as_mut().unwrap().mut_context().set_term(term);
         }
-        let readcmd = cmd.readonly();
-        let tag = cmd.tag();
-        let pool = channels.take_pool();
-        if readcmd {
+        let pool = executor.take_pool();
```

Comment:
I think it is more better to move following codes into Executor::exec, so the Executor is more like a executor.